From 976648f845bc9677558f9695cc91952e19001dbb Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Wed, 4 May 2016 10:56:40 +0800
Subject: [PATCH 1048/5242] MLK-12748-2 ARM: imx: remove IOMUXC GPR setting
 for i.mx7d TO1.2

commit  455333bc6d8560217cde82f6c68c517a02d3aac7 from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX7D TO1.2 removes the DDR PAD retention mode setting
in IOMUXC GPR, it is same as TO1.0, so only apply the
IOMUXC GPR setting for TO1.1.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/suspend-imx7.S |   12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-imx/suspend-imx7.S b/arch/arm/mach-imx/suspend-imx7.S
index d657b24..744621e 100644
--- a/arch/arm/mach-imx/suspend-imx7.S
+++ b/arch/arm/mach-imx/suspend-imx7.S
@@ -303,9 +303,9 @@
 
 	ldr	r11, [r0, #PM_INFO_MX7_ANATOP_V_OFFSET]
 	ldr	r7, [r11, #ANADIG_DIGPROG]
-	and	r7, r7, #0x11
-	cmp	r7, #0x10
-	beq	10f
+	and	r7, r7, #0xff
+	cmp	r7, #0x11
+	bne	10f
 
 	/* TO 1.1 */
 	ldr	r11, [r0, #PM_INFO_MX7_IOMUXC_GPR_V_OFFSET]
@@ -399,9 +399,9 @@
 	str	r7, [r2, r6]
 
 	ldr	r7, [r1, #ANADIG_DIGPROG]
-	and	r7, r7, #0x11
-	cmp	r7, #0x10
-	beq	12f
+	and	r7, r7, #0xff
+	cmp	r7, #0x11
+	bne	12f
 
 	/*
 	 * TKT262940:
-- 
1.7.9.5

